dolphin/db/api.py

```diff
@@ -47,68 +47,218 @@
+                      paired with corresponding item in sort_dirs
+    :param sort_dirs: list of directions in which results should be sorted,
+                      paired with corresponding item in sort_keys, for example
+                      'desc' for descending order
+    :param filters: dictionary of filters
+    :param offset: number of items to skip
+    :returns: list of storage pools
+    """
+    return IMPL.pool_get_all(context, marker, limit, sort_keys, sort_dirs,
+                             filters, offset)
+
+
+def disk_create(context, values):
+    """Create a disk from the values dictionary."""
+    return IMPL.disk_create(context, values)
 
-def registry_context_create(values):
-    return IMPL.registry_context_create(values)
 
+def disk_update(context, disk_id, values):
+    """Update a disk withe the values dictionary."""
+    return IMPL.disk_create(context, disk_id, values)
 
-def registry_context_update(values):
-    return IMPL.registry_context_update(values)
 
+def disk_get(context, disk_id):
+    """Get a disk or raise an exception if it does not exist."""
+    return IMPL.disk_get(context, disk_id)
 
-def registry_context_get(storage_id):
-    return IMPL.registry_context_get(storage_id)
 
+def disk_get_all(context, marker=None, limit=None, sort_keys=None,
+                 sort_dirs=None, filters=None, offset=None):
+    """Retrieves all disks.
 
-def registry_context_get_all():
-    return IMPL.registry_context_get_all()
+    If no sort parameters are specified then the returned volumes are sorted
+    first by the 'created_at' key and then by the 'id' key in descending
+    order.
+
+    :param context: context of this request, it's helpful to trace the request
+    :param marker: the last item of the previous page, used to determine the
+                   next page of results to return
+    :param limit: maximum number of items to return
+    :param sort_keys: list of attributes by which results should be sorted,
+                      paired with corresponding item in sort_dirs
+    :param sort_dirs: list of directions in which results should be sorted,
+                      paired with corresponding item in sort_keys, for example
+                      'desc' for descending order
+    :param filters: dictionary of filters
+    :param offset: number of items to skip
+    :returns: list of disks
+    """
+    return IMPL.disk_get_all(context, marker, limit, sort_keys, sort_dirs,
+                             filters, offset)
+
+
+def storage_access_create(context, values):
+    """Create a storage access that used to connect
+    to a specific storage device.
+    """
+    return IMPL.storage_access_create(context, values)
+
+
+def storage_access_update(context, storage_access_id, values):
+    """Update a storage access with the values dictionary."""
+    return IMPL.storage_access_update(context, storage_access_id, values)
+
+
+def storage_access_get(context, storage_id):
```

Comment:
Ideally the storage_acess_get is confusing, it seems someone is trying to get the access of storage which means some PUT/POST call and not GET. a we name it as get_storage_access_info, update_storage_access_info or delete_storage_access_info?

Reply:
storage_access has changed to access_info.